plugins/janus_videoroom.c

```diff
@@ -944,21 +969,19 @@ static void janus_videoroom_notify_participants(janus_videoroom_publisher *parti
 }
 
 static void janus_videoroom_leave_or_unpublish(janus_videoroom_publisher *participant, gboolean is_leaving) {
+	/* we need to check if the room still exists, may have been destroyed already */
+	if(!participant->room || participant->room->destroyed)
```

Comment:
Yes, it should be.  About half the tests for `room->destroyed` were not atomic.  I'll change them all.